modules/transactionpool/persist.go

```diff
@@ -0,0 +1,145 @@
+package transactionpool
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/NebulousLabs/Sia/build"
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/persist"
+	"github.com/NebulousLabs/Sia/types"
+
+	"github.com/NebulousLabs/bolt"
+)
+
+var (
+	// bucketRecentConsensusChange holds the most recent consensus change seen
+	// by the transaction pool.
+	bucketRecentConsensusChange = []byte("RecentConsensusChange")
+
+	// bucketConfirmedTransactions holds the ids of every transaction that has
+	// been confirmed on the blockchain.
+	bucketConfirmedTransactions = []byte("ConfirmedTransactions")
+
+	// errNilConsensusChange is returned if there is no consensus change in the
+	// database.
+	errNilConsensusChange = errors.New("no consensus change found")
+
+	// fieldRecentConsensusChange is the field in bucketRecentConsensusChange
+	// that holds the value of the most recent consensus change.
+	fieldRecentConsensusChange = []byte("RecentConsensusChange")
+)
+
+// resetDB deletes all consensus related persistence from the transaction pool.
+func (tp *TransactionPool) resetDB(tx *bolt.Tx) error {
+	err := tx.DeleteBucket(bucketConfirmedTransactions)
+	if err != nil {
+		return err
+	}
+	err = tp.putRecentConsensusChange(tx, modules.ConsensusChangeBeginning)
+	if err != nil {
+		return err
+	}
+	_, err = tx.CreateBucket(bucketConfirmedTransactions)
+	return err
+}
+
+// initPersist creates buckets in the database
+func (tp *TransactionPool) initPersist() error {
+	// Create the persist directory if it does not yet exist.
+	err := os.MkdirAll(tp.persistDir, 0700)
+	if err != nil {
+		return err
+	}
+
+	// Open the database file.
+	tp.db, err = persist.OpenDatabase(dbMetadata, filepath.Join(tp.persistDir, dbFilename))
+	if err != nil {
+		return err
+	}
+
+	// Create the database and get the most recent consensus change.
+	cc := modules.ConsensusChangeBeginning
```

Comment:
In this case the `cc` variable is also being overwritten before it is ever read.